src/openrct2/actions/WallPlaceAction.hpp

```diff
@@ -0,0 +1,651 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2019 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "../OpenRCT2.h"
+#include "../management/Finance.h"
+#include "../ride/RideGroupManager.h"
+#include "../ride/Track.h"
+#include "../ride/TrackData.h"
+#include "../world/Banner.h"
+#include "../world/LargeScenery.h"
+#include "../world/MapAnimation.h"
+#include "../world/Scenery.h"
+#include "../world/SmallScenery.h"
+#include "../world/Surface.h"
+#include "GameAction.h"
+
+DEFINE_GAME_ACTION(WallPlaceAction, GAME_COMMAND_PLACE_WALL, GameActionResult)
+{
+private:
+    int32_t _wallType;
+    CoordsXYZ _loc;
+    uint8_t _edge;
+    int32_t _primaryColour;
+    int32_t _secondaryColour;
+    int32_t _tertiaryColour;
+
+public:
+    WallPlaceAction()
+    {
+    }
+
+    WallPlaceAction(
+        int32_t wallType, CoordsXYZ loc, uint8_t edge, int32_t primaryColour, int32_t secondaryColour, int32_t tertiaryColour)
+        : _wallType(wallType)
+        , _loc(loc)
+        , _edge(edge)
+        , _primaryColour(primaryColour)
+        , _secondaryColour(secondaryColour)
+        , _tertiaryColour(tertiaryColour)
+    {
+    }
+
+    uint16_t GetActionFlags() const override
+    {
+        return GameAction::GetActionFlags();
+    }
+
+    void Serialise(DataSerialiser & stream) override
+    {
+        GameAction::Serialise(stream);
+
+        stream << DS_TAG(_wallType) << DS_TAG(_loc) << DS_TAG(_edge) << DS_TAG(_primaryColour) << DS_TAG(_secondaryColour)
+               << DS_TAG(_tertiaryColour);
+    }
+
+    GameActionResult::Ptr Query() const override
```

Comment:
Should check all parameters in here that are passed.

Reply:
I've added a few extra on _loc for when z is not zero and and additional check on the edge value.